MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+- **Authors**: [Primata](mailto:primata@movementlabs.xyz)
+- **Desiderata**: [MD-21](../MD/md-21)
+
+## Abstract
+
+This proposal advocates for replacing the current HTLC-based bridge design, which requires four transactions and introduces significant complexity, with a simpler and more efficient two-transaction mechanism. The proposed design reduces costs, enhances security, minimizes user friction, and avoids potential exploits caused by refund logic. It addresses long-standing issues such as reliance on sponsored transactions, outdated audits, and unimplemented fee mechanisms speeding the process to achieve a fully functional bridge that movement yet has been unable to complete, and lots of under issues raised and still pending.
+
+## Motivation
+
+The current bridge design poses numerous challenges, including inefficiency, cost, and user frustration:
+
+1. **Transaction Complexity**: Requires four interactions:
+   - Two by the user (initiation and finalization).
+   - One by the relayer to complete the transaction.
+   - One by a third party (tasked by the relayer) to finalize refunds in case of failure.
+   - If any of these fails, users and the protocol face losses.
+2. **High Cost**:
+   - The multi-transaction setup is expensive for both users and the protocol.
+   - Misestimations in fee calculations could result in significant losses.
+   - Example: At 16 gwei, bridging costs $20 for 10k transactions. If fees are miscalculated by 25%, this leads to a $25k loss. At 100 gwei, the loss escalates dramatically.
+3. **Unfriendly User Experience**:
+   - The current design requires users to have funds on the destination chain for finalization. This increases friction and discourages adoption.
+   - Sponsored transactions are essential to the current implementation but remain unimplemented, leaving users stranded without funds.
+4. **Security Risks**:
+   - Relayer keys are a critical point of failure. This assumption cannot be avoided.
+   - Refund logic introduces vulnerabilities where attackers could exploit the rate limit and relayer downtime.
+   - For instance, if the relayer fails to finalize on the initiator, a malicious actor could:
+     - Take over the refund keys.
+     - Exploit both directions of the bridge (e.g., ETH→MOVE and MOVE→ETH) repeatedly, draining funds.
+5. **Maintenance Burden**:
+   - The current audit is outdated and does not reflect the significant changes made since.
+   - The current design requires a complete UI/UX overhaul, adding complexity and delay.
+6. **Unnecessary Complexity**:
+   - HTLC-based bridges are largely abandoned in favor of simpler, more effective designs.
+   - Examples like the [Consensys HTLC](https://github.com/Consensys/htlc-bridge) bridge demonstrate the [pitfalls of such approaches](https://entethalliance.org/crosschain-bridges-overview-where-we-are-now/).
```

Comment:
I don't see how the first link demonstrates the pitfalls of such approaches. 
The first link is outdated (in fact it was not used by Consensys, but rather GPACT was considered, and then abandoned).
The second link [Crosschains, where are we now](https://entethalliance.org/crosschain-bridges-overview-where-we-are-now/) is an overview of the cross chain solutions and their main features. It does not demonstrate the pitfalls of the [Consensys HTLC bridge](https://github.com/Consensys/htlc-bridge).

> This method is trust minimized, but requires both parties to stay online for the duration of **the swap** in order to withdraw funds on the other side, which can cause friction for end-users.
> It seems that each design has some pro and some con.

This applies to an **atomic swap**. It is unclear to me why there was a decision to use a swap architecture to implement a bridge. I mentioned it 3 weeks ago [here](https://github.com/movementlabsxyz/MIP/pull/39#issuecomment-2448794875).